cmd/samAssembler/samAssembler_test.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"os"
+	"testing"
+)
+
+var SamAssemblerTests = []struct {
+	SamFileName          string
+	RefFile              string
+	OutFileA             string
+	OutFileB             string
+	MultiFaDir           string
+	qNameA               string
+	qNameB               string
+	Delta                float64
+	Gamma                float64
+	Epsilon              float64
+	Kappa                float64
+	LikelihoodCacheSize  int
+	SetSeed              int64
+	ExpectedOutFileA     string
+	ExpectedOutFileB     string
+	OutMultiFaFileNames  []string
+	ExpectedMultiFaFiles []string
+	Verbose              int
+}{
+	{SamFileName: "testdata/diverged.RefAln.sorted.bam",
+		RefFile:              "testdata/ref.fa",
+		OutFileA:             "testdata/tmp.OutFileA.fa",
+		OutFileB:             "testdata/tmp.OutFileB.fa",
```

Comment:
If possible, please add a test case with sequences of 20-50 bases (no more than 100), so I can have a simple example to follow the logic of your main function and perform the calculations in my head or by hand